lib/batch.js

```diff
@@ -68,6 +68,23 @@ module.exports = function(config) {
         });
     };
 
+    batch.get = function(ids, dataset, callback) {
+        var results= []
+        ids.forEach(function(id){
+            var key = { dataset: dataset, id: 'id!' + primary };
+            config.dyno.getItem(key, function(err, item) {
```

Comment:
yep -- any of the `dyno` functions are going to be async and you'll need to handle them as such. Usually we use https://github.com/mbostock/queue.

However: there is a [dyno.getItems function](https://github.com/mapbox/dyno/blob/76aaab798bd50d8c1ee4ed6f71f434cde3a27872/lib/batch.js#L11) that takes advantage of a [batchGetItem request](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB.html#batchGetItem-property) that can be made against DynamoDB. This means we sent one payload to DynamoDB, and it manages making all the requests for us.

If you use this you shouldn't even have to loop through the array of ids at all.